Added missing dependency Harfbuzz to GTK3

XaviDCR92 opened the pull request:
`pcsxr` did not build on Kubuntu 20.04 LTS since GTK3 could not find `hb.h`, which is provided by `libharfbuzz-dev`. Since GTK3 seems to have this new dependency, the following CMake files must be updated accordingly.

Commit:
Support for multiple sockets

Commit:
Added dynstr, fixed NULL pointer access

Commit:
Introduced dynstr, other breaking changes

Commit:
Added missing dependency Harfbuzz to GTK3

Commit:
Merge branch 'master' into multisockets

loathingKernel commented:
Does this supersede #16 ?

It is better than the hacky include directive in #16 for sure, but did #16 not fix the problem for you or you didn't test it?

ghost commented:
I was not aware of #16 to be honest, I'm sorry. Coincidentally, the solution on #16 was the first thing I came up with until I realized other dependencies were using separate `.cmake` files, yet fortunately it was easy to set it up for `harfbuzz`.
OTOH, as you can see I did not include the `set(OpenGL_GL_PREFERENCE LEGACY)` thing since I could compile `pcsxr` fine without it, but of course I agree with that change.

loathingKernel commented:
I think this is a better way to handle the `HarfBuzz` dependency. I will close the other one and create a separate one for the OpenGL directive after I test it on my machine.

Commit:
client_socket was being accidentally reset to -1

That was making the telnet debug server to crash, so fortunately that
solved it while also having the neat features from dynstr.

Commit:
Added GDB server on CpuCfg UI dialog, rearranged layout

The layout was rearranged to make it fit on smaller screens
e.g.: 768pix height as featured on many laptops.
GDB server not implemented yet.

Commit:
Started implementing GDB server

Commit:
Replaced in-house gdb stub by MIT-licensed implementation

Imported from a fork of https://github.com/mborgerson/gdbstub

Commit:
Implemented port-specific wrappers and logic for gdb stub

Commit:
Added "-gdb" command line argument to start gdb stub

Commit:
OpenGL_GL_PREFERENCE set to LEGACY

Commit:
OpenGL_GL_PREFERENCE set to LEGACY

Commit:
Merge branch 'multisockets'

Commit:
Updated gdbstub

Commit:
Make execI() wait until queue is initialized

Commit:
gdb halt command was not being sent to CPU interpreter

Commit:
Attempt to make gdb server to exit gracefully

caiotbc commented:
I'm still unable to compile due to the hb.h not found error. #16 didn't fix it for me, how can I solve this issue? Ubuntu 20.04 here

ghost commented:
Hi @caiotbc , have you ensured `libharfbuzz-dev` is installed on your system?

caiotbc commented:
Yes, `sudo apt install libharfbuzz-dev` says `libharfbuzz-dev is already the newest version (2.6.4-1ubuntu4).`
What other info about my system is relevant to share?

ghost commented:
@caiotbc have you ensured 90fb1eb4 is applied? Take into account it has not been merged into upstream yet, so it is only available from my fork.

caiotbc commented:
Hi, thanks for replying. I've tried cloning from your fork, but now cmake fails with `Cannot find source file:gdbstub/gdbstub.c` and `Cannot find source file:dynstr/dynstr.c`. What are those files and what can I do to find them?

ghost commented:
@caiotbc Those are new dependencies needed by the fork. `dynstr` is used to replace a large static buffer used by `debug.c`, whereas `gdbstub` is, well, a modified implementation of a GDB stub that allows source-level debugging. The latter is a rather cool feature that can be useful for those still writing homebrew games for PS1 using modern versions of the GNU toolchain e.g.: [PSXSDK](https://github.com/XaviDCR92/psxsdk) or [PSn00bSDK](https://github.com/Lameguy64/PSn00bSDK). It's only currently available for POSIX-compliant operating systems e.g.: GNU/Linux though - a Windows port is still on the way.

Actually, https://github.com/iCatButler/pcsxr/commit/90fb1eb4e78a84649ac9e7a75e585e689f74ec31 does not depend on these submodules, so you can move your `HEAD` to the commit via `git checkout 90fb1eb` and go ahead without them. Otherwise, run `git submodule update --init --recursive` on the project root folder and they should download.

caiotbc commented:
@XaviDCR92 Thanks for your help and detailed explanation! I ran `git submodule update --init --recursive` and it compiled successfully. The emulator is working fine, tested with ace combat 2.

ghost commented:
@iCatButler , could 90fb1eb4e78a84649ac9e7a75e585e689f74ec31 be finally merged into upstream?

Commit:
Do not continue program on add/remove breakpoint

Commit:
Set sockets to invalid value on init

Commit:
Make socket blocking on dbg_sys_getc

Commit:
Force Config.Debug=0 on GDB server enabled

Commit:
Add -gdb flag into documentation

Commit:
Call perror on socket.c

Commit:
Use select(2) when accessing several file descriptors at once

Commit:
Notify gdb stub whenever a break instruction is executed

Commit:
Fix typo in string

Commit:
Provisionally redirect RawReadSocket to ReadSocket

Commit:
Consider sockets less than zero as invalid

Commit:
Fix typo on fprintf call

Commit:
Reset Config.GdbServer if GDB server init fails

d33tah commented:
@XaviDCR92 Hi! Are you still working on this patch? I tried to build it and got the following error:

```
-- Configuring done
CMake Error at dynstr/CMakeLists.txt:5 (add_library):
  Cannot find source file:

    dynstr/dynstr.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx


CMake Error at gdbstub/CMakeLists.txt:5 (add_library):
  Cannot find source file:

    gdbstub/gdbstub.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx


CMake Error at dynstr/CMakeLists.txt:5 (add_library):
  No SOURCES given to target: dynstr


CMake Error at gdbstub/CMakeLists.txt:5 (add_library):
  No SOURCES given to target: gdbstub


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

ghost commented:
Hi @d33tah , did you run `git submodule update --init --recursive` after cloning the repository? It should download the external dependencies you are currently missing.
